src/button.c

```diff
@@ -0,0 +1,147 @@
+#include "common.h"
+#include "button.h"
+#include "gpio_rw.h"
+#include "verbose.h"
+
+#include <pthread.h>
+#include <time.h>
+
+extern SNIPS_RUN_PARA RUN_PARA;
+
+static uint8_t flag_long_press = 0;
+static uint8_t flag_short_press = 0;
+static uint8_t current_raw = 1;
+static uint8_t last_raw = 1;
+static time_t start_time = 0;
+static int time_duration = 0;
+static uint8_t long_press_returned = 0;
+
+static pthread_t key_status_observer_daemon;
+
+void (*short_press_callback)(void);
+void (*long_press_callback)(void);
+
+static int button_pin = -1;
+static int button_val = -1;
+
+static void Key_Status_Handler(void){
+    time_t curr_time;
+    time(&curr_time);
+    uint8_t pattern_press = ( !button_val << 1) | button_val;
+    uint8_t pattern_release = ( button_val << 1) | !button_val;
+    uint8_t pattern_hold = ( button_val << 1 ) | button_val;
+
+    time_duration = curr_time - start_time;
+    uint8_t res = (last_raw << 1) | (current_raw);
+
+    if( res == pattern_press ){
```

Comment:
chose one coding style for if then else
and there should not have space after "(" and before ")"